Sources/PowerSync/Kotlin/KotlinPowerSyncDatabaseImpl.swift

```diff
@@ -165,38 +165,42 @@ final class KotlinPowerSyncDatabaseImpl: PowerSyncDatabaseProtocol {
         parameters: [Any]?,
         mapper: @escaping (SqlCursor) -> RowType
     ) throws -> AsyncThrowingStream<[RowType], Error> {
-        AsyncThrowingStream { continuation in
-            Task {
-                do {
-                    for await values in try self.kotlinDatabase.watch(
-                        sql: sql,
-                        parameters: parameters,
-                        mapper: mapper
-                    ) {
-                        try continuation.yield(safeCast(values, to: [RowType].self))
-                    }
-                    continuation.finish()
-                } catch {
-                    continuation.finish(throwing: error)
-                }
-            }
-        }
+        try watch(options: WatchOptions(sql: sql, parameters: parameters, mapper: mapper))
     }
 
     func watch<RowType>(
         sql: String,
         parameters: [Any]?,
         mapper: @escaping (SqlCursor) throws -> RowType
+    ) throws -> AsyncThrowingStream<[RowType], Error> {
+        try watch(options: WatchOptions(sql: sql, parameters: parameters, mapper: mapper))
+    }
+
+    func watch<RowType>(
+        options: WatchOptions<RowType>
     ) throws -> AsyncThrowingStream<[RowType], Error> {
         AsyncThrowingStream { continuation in
             Task {
                 do {
                     var mapperError: Error?
+                    // HACK!
```

Comment:
My preferred solution for this would be to add a utility wrapping errors to our SDK, e.g. in `iosMain` of the `core` package:

```Kotlin
package com.powersync

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.catch
import kotlinx.coroutines.flow.map

public val <T>  Flow<T>.asResult: Flow<Result<T>> get() = this
    .map(Result.Companion::success)
    .catch { error -> emit(Result.failure(error)) }
```

Then, we could use that as a catchall in the Swift SDK when we're listening on Flows.